navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -136,48 +142,160 @@ private static void putLegInformation(ObjectNode legJson, ResponsePath path, int
+                    // oops, we expected it to have a bearing of 90
+                    continue;
                 }
-                // Bearings
-                List<Integer> bearingsList = (List<Integer>) intersectionValue.getOrDefault("bearings", Collections.emptyList());
-                ArrayNode bearingsrray = intersection.putArray("bearings");
-                for (Integer bearing : bearingsList) {
-                    bearingsrray.add(bearing);
+
+                List<Integer> mergedBearings = new ArrayList<>();
+                List<Boolean> mergedEntry = new ArrayList<>();
+
+                // We copy over all bearings and entries for all edges but for the Outgoing Edge
+                int mergedIn = -1;
+                for (int b = 0; b < bearings.size(); b++) {
+                    if (b == out) {
+                        continue;
+                    }
+                    mergedBearings.add(bearings.get(b));
+                    mergedEntry.add(entry.get(b));
+                    if (b == in) {
+                        mergedIn = mergedBearings.size() - 1;
+                    }
                 }
-                // in
-                if (intersectionValue.containsKey("in")) {
-                    intersection.put("in", (int) intersectionValue.get("in"));
+
+                Map<String, Object> nextIntersection = (Map<String, Object>) list.get(i + 1).getValue();
+                List<Integer> nextBearings = (List<Integer>) nextIntersection.get("bearings");
+                Integer nextIn = (Integer) nextIntersection.get("in");
+                Integer nextOut = (Integer) nextIntersection.get("out");
+                List<Boolean> nextEntry = (List<Boolean>) nextIntersection.get("entries");
+
+                if (nextBearings.get(nextIn) != 90) {
+                    // oops, it should have a bearing of 90
+                    continue;
                 }
-                // out
-                if (intersectionValue.containsKey("out")) {
-                    intersection.put("out", (int) intersectionValue.get("out"));
+                // We copy over all bearings and entries for all edges but for the Incoming
+                int mergedOut = -1;
+                for (int b = 0; b < nextBearings.size(); b++) {
+                    if (b == nextIn) {
+                        continue;
+                    }
+                    mergedBearings.add(nextBearings.get(b));
+                    mergedEntry.add(nextEntry.get(b));
+                    if (b == nextOut) {
+                        mergedOut = mergedBearings.size() - 1;
+                    }
                 }
+
+                // remove the duplicate Intersection
+                list.remove(i + 1);
+                Map<String, Object> mergedIntersection = new HashMap<>();
+                mergedIntersection.put("in", mergedIn);
+                mergedIntersection.put("out", mergedOut);
+                mergedIntersection.put("bearings", mergedBearings);
```

Comment:
While I personally don't care, Mapbox is always ordering the bearings clockwise from 0 - 359.  I am not sure if they rely on this order for something internally.

Reply:
fixed